src/Microsoft.AspNetCore.JsonPatch/JsonPatchDocumentOfT.cs

```diff
@@ -705,61 +705,57 @@ IList<Operation> IJsonPatchDocument.GetOperations()
             return allOps;
         }
 
-        private string GetPath<TProp>(Expression<Func<TModel, TProp>> expr)
+        // Internal for testing
+        internal string GetPath<TProp>(Expression<Func<TModel, TProp>> expr, string position)
         {
-            return "/" + GetPath(expr.Body, true).ToLowerInvariant();
+            var segments = GetPathSegments(expr.Body);
+            var path = String.Join("/", segments);
+            if (position != null)
+            {
+                path += "/" + position;
+                if (segments.Count == 0)
+                {
+                    return path.ToLowerInvariant();
+                }
+            }
```

Comment:
what about a case where `segments.Count == 0` and `position != null`? It looks like the value would be missing the leading slash.

Reply:
A good way to think about this is that `path = "/" + string.Join("/", segments)` will always generate the right thing even when segments is empty.

Then you just have to deal with appending the position.

Reply:
This case is covered. Adding tests where position is not null